cpp/src/Slice/Util.h

```diff
@@ -49,8 +49,8 @@ std::vector<std::string>
 splitScopedName(const std::string&);
 
 // Checks an identifier for illegal syntax and reports any that is present.
-void
-checkIdentifier(const std::string&);
+bool
+checkIdentifier(const std::string&, const std::string& = "");
```

Comment:
the second parameter should not have a default value.

Reply:
`checkIdentifier` was used in some places before my changes, and in those places there's only one parameter, such as:
https://github.com/zeroc-ice/ice/blob/640e8e0a42c48f10770925782e01fdbdd54ac321/cpp/src/Slice/Grammar.y#L548-L559
The default value is there so it can be used in place of `isLegalName`, without breaking the old places it used to be used.